gnomad_qc/v3/sample_qc/sample_qc.py

```diff
@@ -334,86 +496,351 @@ def apply_stratified_filters(filtering_qc_metrics: List[str]) -> hl.Table:
+
+
+def get_relationship_filter_expr(
+    hard_filtered_expr: hl.expr.BooleanExpression,
+    relationship: str,
+    relationship_set: hl.expr.SetExpression,
+) -> hl.expr.builders.CaseBuilder:
+    """
+    Returns case statement to populate relatedness filters in sample_filters struct
+    :param hl.expr.BooleanExpression hard_filtered_expr: Boolean for whether sample was hard filtered.
+    :param str relationship: Relationship to check for. One of DUPLICATE_OR_TWINS, PARENT_CHILD, or SIBLINGS.
+    :param hl.expr.SetExpression relationship_set: Set containing all possible relationship strings for sample.
+    :return: Case statement used to population sample_filters related filter field.
+    :rtype: hl.expr.builders.CaseBuilder
+    """
+    return (
+        hl.case()
+        .when(hard_filtered_expr, hl.null(hl.tbool))
+        .when(hl.is_defined(relationship_set), relationship_set.contains(relationship))
+        .default(False)
+    )
+
+
+def compare_row_counts(ht1: hl.Table, ht2: hl.Table) -> bool:
+    """
+    Checks if row counts in two Tables are the same
+    :param Table ht1: First Table to be checked
+    :param Table ht2: Second Table to be checked
+    :return: Whether the row counts are the same
+    :rtype: bool
+    """
+    r_count1 = ht1.count()
+    r_count2 = ht2.count()
+    logger.info(f"{r_count1} rows in left table; {r_count2} rows in right table")
+    return r_count1 == r_count2
+
+
+def join_tables(
+        left_ht: hl.Table, left_key: str, right_ht: hl.Table, right_key: str, join_type: str
+) -> hl.Table:
+    """
+    Joins left and right tables using specified keys and join types and returns result.
+    Also prints warning if sample counts are not the same.
+    :param Table left_ht: Left Table to be joined
+    :param str left_key: Key of left Table
+    :param Table right_ht: Right Table to be joined
+    :param str right_key: Key of right Table
+    :param str join_type: Type of join
+    :return: Table with annotations
+    :rtype: Table
+    """
+    sample_count_match = compare_row_counts(left_ht, right_ht)
+    if not sample_count_match:
+        logger.warning("Sample counts in left and right tables do not match!")
+    return left_ht.key_by(left_key).join(right_ht.key_by(right_key), how=join_type)
+
+
+def generate_metadata(regressed_metrics_outlier: bool = True) -> hl.Table:
+    """
+    Pulls all sample QC information together in one Table
+
+    :param regressed_metrics_outlier: Should the outlier table be from the residuals instead of pop stratified
+    :return:
+    """
+    logging_statement = "Reading in {} and joining with meta HT"
+
+    logger.info("Loading metadata file with subset, age, and releasable information to begin creation of the meta HT")
+    left_ht = get_gnomad_v3_mt(remove_hard_filtered_samples=False).cols()
+    right_ht = project_meta.ht()
+    right_ht = right_ht.annotate(exclude=hl.if_else(hl.is_missing(right_ht.exclude), False, right_ht.exclude))
```

Comment:
A style choice, I think keeping the `gnomad` and `project_meta` names or adding it into the variable name might be easier to track the joins that are done below. I've been scrolling back up to remember which table is which.

Reply:
I think I will keep as is, otherwise all of the `left_ht` and `right_ht` will need to be renamed and I think it adds a lot of variables that are not used much, but if you both think that would be more readable I am happy to change them all